### ISSUE TITLE
typescript gets included in node_modules

### ISSUE THREAD

**Issue by serg06** [serverless/serverless-plugin-typescript#252]
- In `yarn.lock`, no package has typescript as a dependency
- In `package.json` it's under `devDependencies`
- In `serverless.yml` I have `excludeDevDependencies: true`

Yet for some reason when I run `sls package`, it's included under `node_modules/typescript` (and it's 60MB large!)

Tried with versions:
- 1.1.9
- 1.2.0
- 2.0.0

**Comment by medikoo** [comment]
@serg06 Are you stating that you have `typescript` listed in service `devDependencies`, yet it gets packaged for lambda (even though exclusion of dev dependencies is applied) ?

**Comment by serg06** [comment]
@medikoo Yes exactly.

**Comment by medikoo** [comment]
@serg06 will it happen also if you run `sls package` but with no `plugins` section in `serverless.yml` (I know it will create not working artifact, but I want to confirm whether any plugin influences that)

**Comment by tusbar** [comment]
@medikoo I can confirm that `typescript` is still included in the package, with `excludeDevDependencies: true`, using serverless ^3.15. No extra plugin.

**Comment by gualopezb** [comment]
Is there any solution to this?

**Comment by serg06** [comment]
IIRC I fixed it by adding this to serverless.yaml:

```yml
package:
  patterns:
    - '!node_modules/typescript/**'
```

Does that work? @gualopezb

**Comment by gualopezb** [comment]
Yes, it does the trick. I was just wondering if this is indeed an issue with the `serverless-plugin-typescript` module and, if so, is it going to be fixed in a future version?
